test-utils/runtime-tester/src/fuzzing.rs

```diff
@@ -140,32 +173,206 @@ impl TransactionConfig {
+                    code: scope.available_contracts[contract_id].code.clone(),
+                })],
+            })
+        });
+
+        // Multiple function calls
+        options.push(|u, scope| {
+            let nonce = scope.nonce();
+
+            let signer_account = scope.random_account(u)?;
+            let receiver_account = scope.random_account(u)?;
+
+            let signer = InMemorySigner::from_seed(
+                signer_account.id.clone(),
+                KeyType::ED25519,
+                signer_account.id.as_ref(),
+            );
+
+            let mut receiver_functions = vec![];
+            for contract_id in receiver_account.deployed_contracts {
+                for function in &scope.available_contracts[contract_id].functions {
+                    receiver_functions.push(function);
+                }
+            }
+
+            if receiver_functions.is_empty() {
+                return Ok(TransactionConfig {
+                    nonce,
+                    signer_id: signer_account.id.clone(),
+                    receiver_id: receiver_account.id.clone(),
+                    signer,
+                    actions: vec![],
+                });
+            }
+
+            let mut actions = vec![];
+            let mut actions_num = u.int_in_range(0..=MAX_ACTIONS)?;
+            if u.int_in_range(0..=10)? == 0 {
+                actions_num = 1;
+            }
+
+            while actions.len() < actions_num && u.len() > Function::size_hint(0).1.unwrap() {
+                let function = u.choose(&receiver_functions)?;
+                actions.push(Action::FunctionCall(function.arbitrary(u)?));
+            }
+
+            Ok(TransactionConfig {
+                nonce,
+                signer_id: signer_account.id.clone(),
+                receiver_id: receiver_account.id.clone(),
+                signer,
+                actions,
+            })
+        });
         let f = u.choose(&options)?;
         f(u, scope)
     }
 
     fn size_hint(_depth: usize) -> (usize, Option<usize>) {
-        (7, Some(14))
+        (7, Some(210))
     }
 }
 
 #[derive(Clone)]
 pub struct Scope {
     accounts: Vec<Account>,
+    alive_accounts: HashSet<usize>,
     nonce: Nonce,
     height: BlockHeight,
+    available_contracts: Vec<Contract>,
+    last_tx_num: usize,
 }
 
 #[derive(Clone)]
 pub struct Account {
     pub id: AccountId,
     pub balance: Balance,
+    pub deployed_contracts: HashSet<ContractId>,
+}
+
+#[derive(Clone)]
+pub struct Contract {
+    pub code: Vec<u8>,
+    pub functions: Vec<Function>,
+}
+
+#[derive(Clone)]
+pub enum Function {
+    StorageUsage,
+    BlockIndex,
+    BlockTimestamp,
+    PrepaidGas,
+    RandomSeed,
+    PredecessorAccountId,
+    SignerAccountPk,
+    SignerAccountId,
+    CurrentAccountId,
+    AccountBalance,
+    AttachedDeposit,
+    ValidatorTotalStake,
+    ExtSha256,
+    UsedGas,
+    WriteKeyValue,
+    WriteBlockHeight,
```

Comment:
Is it intentional that `promise_*` host functions are not included? Also I wonder whether there is a way to import the host functions we have in vm logic directly to make sure that the set of host functions we fuzz is the same as the ones we actually use. cc @matklad

Reply:
Yes, it is intentional in two ways. First, I don't want to add promises in this stage, because it is a bit complex and adds weight to the review. Second, I was going by implemented functions in test_contract, I'm not calling host functions directly (I'm not sure if I can, maybe someone can explain to me how to do it correctly). In test contract only one function uses promises and it is, again, way to big to be included in this review.